cluster/spanning_tree_main.cc

```diff
@@ -17,33 +17,70 @@ int getpid() { return (int)::GetCurrentProcessId(); }
 
 #include <iostream>
 #include <fstream>
-#include <string.h>
-#include <stdlib.h>
-#include <errno.h>
 
 using namespace VW;
 
+#include <boost/program_options.hpp>
+namespace po = boost::program_options;
+
+void usage(const po::options_description& desc)
+{
+  std::cout << "usage: spanning_tree [--port,-p number] [--nondaemon] [--help,-h] [pid_file]" << std::endl;
+  std::cout << desc << std::endl;
+}
+
 int main(int argc, char* argv[])
 {
-  if (argc > 2)
+  int port = 26543;
```

Comment:
has this always been the default port?

Reply:
yep, its in the spanning tree constructor. its also hard coded into vw

Reply:
should it live as a constexpr somewhere then and be re-used?

Reply:
This binary does not depend on VW at all. Maybe in future. Common lib constants or something?

Reply:
mmm ok I see